src/main/java/ay2122s1_cs2103t_w16_2/btbb/logic/descriptors/OrderDescriptor.java

```diff
@@ -1,38 +1,93 @@
 
+import ay2122s1_cs2103t_w16_2.btbb.commons.core.index.Index;
+import ay2122s1_cs2103t_w16_2.btbb.exception.CommandException;
+import ay2122s1_cs2103t_w16_2.btbb.model.Model;
+import ay2122s1_cs2103t_w16_2.btbb.model.client.Address;
+import ay2122s1_cs2103t_w16_2.btbb.model.client.Client;
+import ay2122s1_cs2103t_w16_2.btbb.model.client.Name;
 import ay2122s1_cs2103t_w16_2.btbb.model.client.Phone;
 import ay2122s1_cs2103t_w16_2.btbb.model.order.Order;
 
 public class OrderDescriptor {
-    private Phone phone;
+    public static final String MESSAGE_MISSING_CLIENT_DETAILS = "Both client and client details cannot be found";
+
+    private Index clientIndex;
+    private Phone clientPhone;
+    private Name clientName;
+    private Address clientAddress;
 
     public OrderDescriptor() {};
 
+    /**
+     * Constructs an {@code OrderDescriptor} using the details of an existing {@code OrderDescriptor}.
+     *
+     * @param toCopy Existing {@code OrderDescriptor}.
+     */
     public OrderDescriptor(OrderDescriptor toCopy) {
-        setPhone(toCopy.phone);
+        setClientIndex(toCopy.clientIndex);
+        setClientPhone(toCopy.clientPhone);
+        setClientName(toCopy.clientName);
+        setClientAddress(toCopy.clientAddress);
     }
 
-    public void setPhone(Phone phone) {
-        this.phone = phone;
+    public void setClientPhone(Phone clientPhone) {
+        this.clientPhone = clientPhone;
     }
 
-    public Optional<Phone> getPhone() {
-        return Optional.ofNullable(phone);
+    public Optional<Phone> getClientPhone() {
+        return Optional.ofNullable(clientPhone);
+    }
+
+    public void setClientIndex(Index clientIndex) {
+        this.clientIndex = clientIndex;
+    }
+
+    public Optional<Index> getClientIndex() {
+        return Optional.ofNullable(clientIndex);
+    }
+
+    public void setClientName(Name clientName) {
+        this.clientName = clientName;
+    }
+
+    public Optional<Name> getClientName() {
+        return Optional.ofNullable(clientName);
+    }
+
+    public void setClientAddress(Address clientAddress) {
+        this.clientAddress = clientAddress;
+    }
+
+    public Optional<Address> getClientAddress() {
+        return Optional.ofNullable(clientAddress);
     }
 
     /**
      * Converts an Order Descriptor to an Order model type.
-     * All non null fields must be present before conversion.
      *
-     * @return {@code Order}.
+     * @param model To get the client list.
+     * @return {@code Order}
+     * @throws CommandException if both the client and its details are missing.
      */
-    public Order toModelType() {
-        requireAllNonNull(phone);
-        return new Order(phone);
+    public Order toModelType(Model model) throws CommandException {
+        List<Client> clientList = model.getFilteredClientList();
+        Optional<Client> client = getClientIndex().isPresent()
+                ? Optional.of(clientList.get(clientIndex.getZeroBased()))
```

Comment:
You need to handle the case when clientIndex is out of bounds. You can refer to `DeleteClientCommand` to see how it is done. Otherwise, the app throws an uncaught exception.